src/main/java/at/hannibal2/skyhanni/features/fishing/TotemOfCorruption.kt

```diff
@@ -0,0 +1,155 @@
+import at.hannibal2.skyhanni.utils.getLorenzVec
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.entity.item.EntityArmorStand
+import net.minecraft.util.EnumParticleTypes
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+private val config get() = SkyHanniMod.feature.fishing.totemOfCorruption
+
+private var display = emptyList<String>()
+private var totems: List<Totem> = emptyList()
+
+class TotemOfCorruption {
+
+    private val group = RepoPattern.group("features.fishing.totemofcorruption")
+    private val totemNamePattern by group.pattern(
+        "totemname",
+        "§5§lTotem of Corruption"
+    )
+    private val timeRemainingPattern by group.pattern(
+        "timeremaining",
+        "§7Remaining: §e(?:(?<min>\\d+)m )?(?<sec>\\d+)s"
+    )
+    private val ownerPattern by group.pattern(
+        "owner",
+        "§7Owner: §e(?<owner>.+)"
+    )
+
+    @SubscribeEvent
+    fun onRender(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isOverlayEnabled() || display.isEmpty()) return
+        config.position.renderStrings(display, posLabel = "Totem of Corruption")
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!event.repeatSeconds(2)) return
+        if (!isOverlayEnabled()) return
+
+        totems = getTotems()
+            .filterNotNull()
+            .mapNotNull { totem ->
+                val timeRemaining = getTimeRemaining(totem)
+                val owner = getOwner(totem)
+                if (timeRemaining != null && owner != null) {
+                    if (
+                        timeRemaining == config.warnWhenAboutToExpire.toDuration(DurationUnit.SECONDS)
+                        && config.warnWhenAboutToExpire.toDuration(DurationUnit.SECONDS) > 0.seconds
+                    ) {
+                        playPlingSound()
+                        sendTitle("§c§lTotem of Corruption §eabout to expire!", 5.seconds)
+                    }
+                    Totem(totem.getLorenzVec(), timeRemaining, owner)
+                } else {
+                    null
+                }
+            }
+
+        display = createDisplay()
+    }
+
+    @SubscribeEvent
+    fun onChatPacket(event: ReceiveParticleEvent) {
+        if (!isHideParticlesEnabled()) return
+
+        for (totem in totems) {
+            if (event.type == EnumParticleTypes.SPELL_WITCH && event.speed == 0.0f) {
+                if (totem.location.distance(event.location) < 4.0) {
+                    event.isCanceled = true
+                }
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onRenderWorld(event: LorenzRenderWorldEvent) {
+        if (!isEffectiveAreaEnabled()) return
+
+        val color = Color(SpecialColour.specialToChromaRGB(config.color), true)
```

Comment:
use this utility function instead
```suggestion
        val color = config.color.toChromaColor()
```